src/SleighAsm.cpp

```diff
@@ -0,0 +1,305 @@
+	s >> alignment;
+}
+
+void SleighAsm::parse_proc_config(DocumentStorage &store)
+{
+	const Element *el = store.getTag("processor_spec");
+	if(!el)
+		throw LowlevelError("No processor configuration tag found");
+
+	const List &list(el->getChildren());
+	List::const_iterator iter;
+
+	for(iter = list.begin(); iter != list.end(); iter++)
+	{
+		const string &elname((*iter)->getName());
+		if(elname == "context_data")
+			context.restoreFromSpec(*iter, &trans);
+	}
+}
+
+void SleighAsm::build_specfile(DocumentStorage &store)
+{
+	const LanguageDescription &language(description[languageindex]);
+	//std::string compiler = archid.substr(archid.rfind(':')+1);
+	//const CompilerTag &compilertag( language.getCompiler(compiler));
+
+	std::string processorfile;
+	//std::string compilerfile;
+	std::string slafile;
+
+	specpaths.findFile(processorfile, language.getProcessorSpec());
+	//specpaths.findFile(compilerfile,compilertag.getSpec());
+	specpaths.findFile(slafile, language.getSlaFile());
+
+	try
+	{
+		Document *doc = store.openDocument(processorfile);
+		store.registerTag(doc->getRoot());
+	}
+	catch(XmlError &err)
+	{
+		ostringstream serr;
+		serr << "XML error parsing processor specification: " << processorfile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	}
+	catch(LowlevelError &err)
+	{
+		ostringstream serr;
+		serr << "Error reading processor specification: " << processorfile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	}
+
+	/*
+	try {
+		Document *doc = store.openDocument(compilerfile);
+		store.registerTag(doc->getRoot());
+	} catch(XmlError &err) {
+		ostringstream serr;
+		serr << "XML error parsing compiler specification: " << compilerfile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	} catch(LowlevelError &err) {
+		ostringstream serr;
+		serr << "Error reading compiler specification: " << compilerfile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	}
+	*/
+
+	try
+	{
+		Document *doc = store.openDocument(slafile);
+		store.registerTag(doc->getRoot());
+	}
+	catch(XmlError &err)
+	{
+		ostringstream serr;
+		serr << "XML error parsing SLEIGH file: " << slafile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	}
+	catch(LowlevelError &err)
+	{
+		ostringstream serr;
+		serr << "Error reading SLEIGH file: " << slafile;
+		serr << "\n " << err.explain;
+		throw SleighError(serr.str());
+	}
+}
+
+void SleighAsm::resolve_arch(const string &archid)
+{
+	std::string baseid = archid.substr(0, archid.rfind(':'));
+	languageindex = -1;
+	for(size_t i = 0; i < description.size(); i++)
+	{
+		if(description[i].getId() == baseid)
+		{
+			languageindex = i;
+			if(description[i].isDeprecated())
+				throw LowlevelError("Language " + baseid + " is deprecated");
+			break;
+		}
+	}
+
+	if(languageindex == -1)
+		throw LowlevelError("No sleigh specification for " + baseid);
+}
+
+void SleighAsm::scan_sleigh(const string &rootpath)
+{
```

Comment:
Since a lot of code here has been copied directly from `SleighArchitecture`, it will be good to have comments for each respective code part explaining the original place where it came from and why it had to be copied.